.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -5,32 +5,28 @@
 - [ ] Did you delete any extraneous printlns/debugging code?
 - [ ] Did you specify the type of improvement?
 - [ ] Did you add appropriate documentation in `docs/src`?
-- [ ] Did you state the API impact?
-- [ ] Did you specify the code generation impact?
 - [ ] Did you request a desired merge strategy?
 - [ ] Did you add text to be included in the Release Notes for this change?
 
+<!--
+If you PR has any impact on the user API or affects backend code generation,
+please describe the change in the "Release Notes" section below.
+-->
+
 #### Type of Improvement
 
 <!-- Choose one or more from the following:
-- bug fix
-- performance improvement
-- documentation
-- code refactoring
-- code cleanup
-- backend code generation
-- new feature/API
+- Feature (or new API)
+- API modification
+- API deprecation
+- Backend code generation
+- Performance improvement
+- Bugfix
+- Documentation or website-related
+- Dependency update
+- Internal or build-related (includes code refactoring/cleanup)
 -->
 
-#### API Impact
-
-<!-- How would this affect the current API? Does this add, extend, deprecate, remove, or break any existing API? -->
-
-#### Backend Code Generation Impact
-
-<!-- Does this change any generated Verilog?  -->
-<!-- How does it change it or in what circumstances would it?  -->
-
 #### Desired Merge Strategy
```

Comment:
@mwachs5 I've made some more changes to the template to streamline it a little and nudge PR authors more toward writing release notes than putting things in these sections (which are/were not included in the release notes).